mkosi/config.py

```diff
@@ -3777,26 +3781,31 @@ def match_config(self, path: Path) -> bool:
 
         return match_triggered is not False
 
-    def parse_config_one(self, path: Path, profiles: bool = False, local: bool = False) -> bool:
+    def parse_config_one(self, path: Path, parse_profiles: bool = False, parse_local: bool = False) -> bool:
         s: Optional[ConfigSetting]  # Make mypy happy
         extras = path.is_dir()
 
         if path.is_dir():
-            path = path / "mkosi.conf"
+            path /= "mkosi.conf"
 
         if not self.match_config(path):
             return False
 
         if extras:
-            if local and (path.parent / "mkosi.local.conf").exists():
-                self.parse_config_one(path.parent / "mkosi.local.conf")
-
-                # Configuration from mkosi.local.conf should override other file based configuration but not
-                # the CLI itself so move the finalized values to the CLI namespace.
-                for s in SETTINGS:
-                    if hasattr(self.config, s.dest):
-                        setattr(self.cli, s.dest, self.finalize_value(s))
-                        delattr(self.config, s.dest)
+            if parse_local:
+                if (
+                    (localpath := path.parent / "mkosi.local/mkosi.conf").exists() or
+                    (localpath := path.parent / "mkosi.local.conf").exists()
```

Comment:
I assume you want the alignment, but ruff following PEP8 would put it like this
```suggestion
                    (localpath := path.parent / "mkosi.local/mkosi.conf").exists()
                    or (localpath := path.parent / "mkosi.local.conf").exists()
```